---
title: "T-cell ligand analysis for activated myofibroblasts"
author: "Jan Lanzer"
date: "4/16/2021"
output:
  html_document: default
  pdf_document: default
editor_options:
  chunk_output_type: console
---

#### Collaborator : 
Florian Sicklinger from Florian Leuschner lab, HD

#### Background:
* CTLA4 is expressed on activated T-cells
* CTLA4 inhibition (Abatacept) caused post MI ventricle rupture in patient 
* the same effect was reproduced in a mouse MI model

#### Hypothesis: 
T-cell activation mediates crucial myofibroblast activation for short term remodeling post MI

#### Data set:
scRNAseq (PMID: 32130914) [already QCed and integrated from other Myofib project]

#### Analysis:
* identification of activated fibroblasts and activated T-cells
* identification of marker genes for both groups
* Performing Nichenet analysis to identify ligands that 
  1) are upregulated in T-cells, 
  2) explain well regulation of genes in activated Fibroblasts
* Perform statistical test to see if genes of interest (act_fibs) are among the top predicted
 genes based on the ligand-target network

#### Open Questions: 
* potential furhter analysis with other tools? Daniel showed that resource has higher impact than the tool so far. As Omnipath networks are "better" curated, I will wait for the bug fix and rerun with Omnipath networks
* Is this study model suited as Ctla4 does not seem to be highly expressed?



```{r setup, include=FALSE, echo= FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	message = TRUE,
	warning = FALSE
)
#library(OmnipathR)
library(nichenetr)
library(tidyverse)
library(Seurat)
library(magrittr)
library(cowplot)
#setwd("~/R-projects/sc-exploration")
source('R-scripts/utils.R')
seu = readRDS('output/cell_mi_files/integrateint_cell_MI_annotated.rds')

```

##Dataset visualization


```{r echo=TRUE, message=TRUE, warning=FALSE}
DimPlot(seu, cols= col.set, label = T)
t.cell = c("Nkg7", "Cd3d", "Ctla4")
DefaultAssay(seu)= "RNA"
FeaturePlot(seu, features = t.cell)
DotPlot(seu, features = t.cell)
## Ctla4 does not seem to be a marker of T-cell activation in this data set..
```

## T-cell Differential Expression Analysis

```{r Get upregulated genes in T-cells after MI}

# How many T-cells are reported by each experimental group?
seu[[]] %>%  
  filter(celltype== "T-cells") %>% 
  count(OP)

# And at each timepoint?
seu[[]] %>%  
  filter(celltype== "T-cells") %>% 
  count(time)

# only 107 cells are from sham operated mice. For the DEA, I will use both none and sham as control
seu$OP_mod= str_replace_all(string = seu$OP, pattern = " ", replacement = ".")
seu$OP_mod= str_replace_all(string = seu$OP_mod, pattern = "sham", replacement = "none")

seu$celltype.stim<- paste(seu$celltype, seu$OP_mod, sep = "_")
unique(seu$celltype.stim)
Idents(seu)= seu$celltype.stim
dea_Tcell=  FindMarkers(seu, ident.1 = "T-cells_myocardial.infarction", ident.2 = "T-cells_none")
up_Tcell= dea_Tcell %>% filter(avg_log2FC >0) %>% arrange(p_val_adj)


# now get markers of act fibroblasts (alternatively load from meta-marker table)
Idents(seu)= seu$celltype
dea_fibs= FindMarkers(seu, ident.1= "fibs_act", ident.2 = c("fibs_1", "fibs_2", "fibs_3"))
up_fibs= dea_fibs %>% filter(avg_log2FC >0) %>% arrange(p_val_adj)


meta_marker =readRDS( file = "output/marker/meta_marker.rds")
act_genes = meta_marker %>%
  filter(study ==  "cell") %>% 
  filter(p_val_adj < 0.05) %>% 
  top_n(100) %>% 
  pull(unique(gene))

```

## Run Nichenet
### Part1 prepare the data
 
```{r run Nichenet - prepare data}
## 1. load networks and ligand_target matrix
ligand_target_matrix = readRDS(url("https://zenodo.org/record/3260758/files/ligand_target_matrix.rds"))
lr_network = readRDS(url("https://zenodo.org/record/3260758/files/lr_network.rds"))
weighted_networks = readRDS(url("https://zenodo.org/record/3260758/files/weighted_networks.rds"))

## 2. convert to mouse symobls

weighted_networks_lr = weighted_networks$lr_sig %>% inner_join(lr_network %>% distinct(from,to), by = c("from","to"))

lr_network = lr_network %>% mutate(from = convert_human_to_mouse_symbols(from), to = convert_human_to_mouse_symbols(to)) %>% drop_na()
colnames(ligand_target_matrix) = ligand_target_matrix %>% colnames() %>% convert_human_to_mouse_symbols()
rownames(ligand_target_matrix) = ligand_target_matrix %>% rownames() %>% convert_human_to_mouse_symbols()

ligand_target_matrix = ligand_target_matrix %>% .[!is.na(rownames(ligand_target_matrix)), !is.na(colnames(ligand_target_matrix))]

weighted_networks_lr = weighted_networks_lr %>% mutate(from = convert_human_to_mouse_symbols(from), to = convert_human_to_mouse_symbols(to)) %>% drop_na()

## 3. define receiver and sender cells and their ligand - target gene sets
# nichenet suggests to use expression cut-offs. I will use the DE-genes from above to in combination with expression cut-off

Idents(seu)= seu$celltype
DefaultAssay(seu)= "RNA"

## receiver cells 

receiver = "fibs_act"
# expressed_genes_receiver = rownames(dea_fibs %>% filter(p_val_adj< 0.05,
#                                                         pct.1> 0.1))
expressed_genes_receiver = get_expressed_genes(receiver, seu, pct = 0.10, assay_oi = "RNA")

background_expressed_genes = expressed_genes_receiver %>% .[. %in% rownames(ligand_target_matrix)]

## sender

# use mabe different 
#sender_celltypes = c("T-cells", "macrophages/monocytes", "granulocytes","B-cells")
sender_celltypes = c("T-cells")

# nichenet suggest to filter by expression: 
list_expressed_genes_sender = sender_celltypes %>% unique() %>% lapply(get_expressed_genes, seu, 0.1, assay_oi= "RNA") # lapply to get the expressed genes of every sender cell type separately here

expressed_genes_sender = list_expressed_genes_sender %>% unlist() %>% unique() 

# add this step to use DE genes in T-cells instead
set1= rownames(up_Tcell %>% filter(p_val_adj<0.05) %>% filter(pct.1 >0.1))
expressed_genes_sender = set1

seurat_obj_receiver= subset(seu, idents = receiver)
seurat_obj_receiver = SetIdent(seurat_obj_receiver, value = seurat_obj_receiver[["OP"]])

geneset_oi = rownames(dea_fibs %>% filter(p_val_adj< 0.05,
                                         pct.1> 0.1))
geneset_oi = geneset_oi %>% .[. %in% rownames(ligand_target_matrix)]

## 4. define a set of potential ligands (expressed by sender, putative receptor expressed by receiver)
ligands = lr_network %>% pull(from) %>% unique()
receptors = lr_network %>% pull(to) %>% unique()

expressed_ligands = intersect(ligands,expressed_genes_sender)
expressed_receptors = intersect(receptors,expressed_genes_receiver)

potential_ligands = lr_network %>% filter(from %in% expressed_ligands & to %in% expressed_receptors) %>% pull(from) %>% unique()

```

### Part2 predict ligands and visualize

```{r run Nichenet - predict activity}

# predict ligand activity
ligand_activities = predict_ligand_activities(geneset = geneset_oi, background_expressed_genes = background_expressed_genes, ligand_target_matrix = ligand_target_matrix, potential_ligands = potential_ligands)

ligand_activities = ligand_activities %>%
   arrange(-pearson) %>% 
  mutate(rank = rank(desc(pearson)))

ligand_activities

## plot - histogram of pearson corr for all ligands
p_hist_lig_activity = ggplot(ligand_activities, aes(x=pearson)) + 
  geom_histogram(color="black", fill="darkorange")  + 
  # geom_density(alpha=.1, fill="orange") +
  geom_vline(aes(xintercept=min(ligand_activities %>% top_n(10, pearson) %>% pull(pearson))), color="red", linetype="dashed", size=1) + 
  labs(x="ligand activity (PCC)", y = "# ligands") +
  theme_classic()

p_hist_lig_activity

# this plot can help to select the best top ligands. Here we only have a few ligands we tested, 
# because of the small overlap of upregulated genes in T-cells AND ligand sin the matrix
# therefore I will use all lignds for downstream analysis, as all ligands display corr>0.1

best_upstream_ligands = ligand_activities %>% 
  #top_n(10, pearson) %>% 
  arrange(-pearson) %>% 
  pull(test_ligand) %>% 
  unique()

# Data transformation for heatmap plotting
active_ligand_target_links_df = best_upstream_ligands %>% lapply(get_weighted_ligand_target_links,geneset = geneset_oi, ligand_target_matrix = ligand_target_matrix, n = 200) %>% bind_rows() %>% drop_na()

active_ligand_target_links = prepare_ligand_target_visualization(ligand_target_df = active_ligand_target_links_df, ligand_target_matrix = ligand_target_matrix, cutoff = 0.33)

order_ligands = intersect(best_upstream_ligands, colnames(active_ligand_target_links)) %>% rev() %>% make.names()
order_targets = active_ligand_target_links_df$target %>% unique() %>% intersect(rownames(active_ligand_target_links)) %>% make.names()
rownames(active_ligand_target_links) = rownames(active_ligand_target_links) %>% make.names() # make.names() for heatmap visualization of genes like H2-T23
colnames(active_ligand_target_links) = colnames(active_ligand_target_links) %>% make.names() # make.names() for heatmap visualization of genes like H2-T23

vis_ligand_target = active_ligand_target_links[order_targets,order_ligands] %>% t()

#  heatmap plotting
p_ligand_target_network = vis_ligand_target %>% make_heatmap_ggplot("Prioritized ligands","Predicted target genes", color = "purple",legend_position = "top", x_axis_position = "top",legend_title = "Regulatory potential")  + theme(axis.text.x = element_text(face = "italic")) + scale_fill_gradient2(low = "whitesmoke",  high = "purple", breaks = c(0,0.0045,0.0090))

p_ligand_target_network

# plotting of ligand expression
Idents(seu)= seu$celltype.stim

p3= DotPlot(subset(seu, idents = "T-cells_myocardial.infarction"), features = best_upstream_ligands %>% rev(), cols = "RdYlBu") +coord_flip()

p1= dea_Tcell[best_upstream_ligands,] %>% 
  #arrange(desc(avg_log2FC)) %>% 
  rownames_to_column("gene")%>%
  drop_na() %>% 
  mutate(dummy= "x")%>%
  ggplot(., aes(x=gene , y= dummy, fill= avg_log2FC))+
  geom_tile()+
  coord_flip()+
  labs(y= "", 
       x= "potential ligands")
p2= dea_Tcell[best_upstream_ligands,] %>% rownames_to_column("gene")%>%
  drop_na() %>% 
  mutate(dummy= "x")%>%
  ggplot(., aes(x=gene , y= dummy, fill= -log(p_val)))+
  geom_tile()+
  coord_flip()+
  labs(y= "", 
       x= "potential ligands")

plot_grid(p1, p2, p3, nrow= 2, rel_widths = c(1,1,1.8))

dea_Tcell["Tgfb1",]
dea_Tcell["Ctla4",]
```

### Part 3 run randomization with background genes

```{r perform randomization to asssess wether the geneset of interest i}
k = 3 # 3-fold
n = 3 # 2 rounds

background_expressed_genes = expressed_genes_receiver %>% .[. %in% rownames(ligand_target_matrix)]
head(background_expressed_genes)

pemt_gene_predictions_top20_list = seq(n) %>% lapply(assess_rf_class_probabilities, 
                                                     folds = k, 
                                                     geneset = geneset_oi, 
                                                     background_expressed_genes = background_expressed_genes, 
                                                     ligands_oi = best_upstream_ligands, 
                                                     ligand_target_matrix = ligand_target_matrix)

target_prediction_performances_cv = pemt_gene_predictions_top20_list %>% 
  lapply(classification_evaluation_continuous_pred_wrapper) %>% 
  bind_rows() %>% 
  mutate(round=seq(1:nrow(.)))


target_prediction_performances_cv$auroc %>% mean()

target_prediction_performances_cv$aupr %>% mean()

target_prediction_performances_cv$pearson %>% mean()


target_prediction_performances_discrete_cv = pemt_gene_predictions_top20_list %>% lapply(calculate_fraction_top_predicted, quantile_cutoff = 0.95) %>% bind_rows() %>% ungroup() %>% mutate(round=rep(1:length(pemt_gene_predictions_top20_list), each = 2))

target_prediction_performances_discrete_cv %>% filter(true_target) %>% .$fraction_positive_predicted %>% mean()

target_prediction_performances_discrete_cv %>% filter(!true_target) %>% .$fraction_positive_predicted %>% mean()

target_prediction_performances_discrete_fisher = pemt_gene_predictions_top20_list %>% lapply(calculate_fraction_top_predicted_fisher, quantile_cutoff = 0.95) 

target_prediction_performances_discrete_fisher %>% unlist() %>% mean()
# significant overrepresentation of genset of interest over background expressed genes in 
# top genes explained by ligands

top_predicted_genes = seq(length(pemt_gene_predictions_top20_list)) %>% lapply(get_top_predicted_genes,pemt_gene_predictions_top20_list) %>% reduce(full_join, by = c("gene","true_target"))
top_predicted_genes %>% filter(true_target) %>% print(n=100)
```

